### ISSUE TITLE
span printout question

### ISSUE THREAD

**Issue by fdncred** [BurntSushi/jiff#87]
First of all, thank you for creating this crate and taking input from twitter/x with how people would like to see a new datetime crate. ❤️ 

I'm exploring jiff in the hopes that it can replace chrono in [nushell](https://github.com/nushell/nushell). I'm working on a nushell plugin to do this. If we set aside the lackluster implementation details of my [dt plugin](https://github.com/fdncred/nu_plugin_dt/blob/main/src/commands/diff.rs), I get a span printed like the example below for trying to find the verbose difference between two datetimes.
```nushell
❯ '2019-05-10T09:59:12-07:00' | dt diff '2024-08-07T09:36:42.367322100-05:00'
P5y2m27dT23h37m30.3673221s
```
What I'm curious about is why it says `5y2m27d` instead of `5y2m3wk6d`? i.e. there is no week component even though there are 27 days. Knowing your work, I can only assume that week is being left out intentionally and left up to me to translate from days to weeks.

**Comment by BurntSushi** [comment]
Yes it is indeed being left out intentionally! This decision comes straight from Temporal. I'm short on time at the moment, so I can't find a precise Temporal reference for you (I'll follow-up probably tomorrow) explaining things, but here's my understanding:

* It is probably not expected in general for weeks to be inserted into durations without them being explicitly requested.
* Weeks are a little bit weirder than other units in that they don't necessarily divide evenly into bigger units. For example, there is always a whole number of days in a month and always a whole number of months in a year. (You'd have to switch to a different calendar, like the ISO 8601 week calendar---which lacks months---to get years that always have a whole number of weeks.)

Note: In non-Gregorian calendars, weeks aren't always 7 days. But this on its own isn't necessarily disqualifying since it just makes it non-uniform, and spans already support other types of non-uniform units.

Fun fact: ISO 8601-1 actually doesn't allow week units to appear with other units. They are only allowed as part of an extension in ISO 8601-2.

I'll dig through the Temporal materials and see if I can fine a more concrete answer. I'm about 95% confident in my above answer though.

> I'm exploring jiff in the hopes that it can replace chrono in [nushell](https://github.com/nushell/nushell).

Please let me know if there's anything I can do to help! Feel free to DM me on Discord or Zulip. I'm burntsushi or burntsushi5 everywhere.
